docs/use/grammar.rst

```diff
@@ -113,6 +113,22 @@ set of supported options and choose their semantics.
         tags: [experimental, d20];  // Associate some tags with the program result.
     end;
 
+Constants
+---------
+
+The ``constants`` block provides a way to declare constants that can be referenced
+in other parts of the script. The syntax for specifying constants is analogous to
+that of a (flat) JSON object or Python dictionary.
```

Comment:
Rather than repeating the text for ``options``, I would suggest just referring readers there:
```suggestion
in other parts of the script. The syntax for specifying constants is the same as
the ``options`` block (up to the opening keyword).
```
Also, I think it's worth specifying _where_ exactly constants can be used:
```suggestion
as parameter values in other parts of the script. The syntax for specifying constants
is the same as the ``options`` block (up to the opening keyword).
```

Reply:
Thanks @Mandrenkov. I'm very slightly in favour of making sure that each section is fully independent. Many users might just look directly at the Constants section without reading the Options section first. That's why I'm repeating the Options description (which I personally find cleaner, as long as the text is more for looking things up, rather than reading from start to finish). I have no strong opinions here though, so I'm going with your latter suggestion.